src/lib/compress.c

```diff
@@ -78,6 +78,10 @@
 #include "sound/compress_params.h"
 #include "sound/compress_offload.h"
 #include "tinycompress/tinycompress.h"
+#include "compress_ops.h"
+#ifdef TINYCOMPRESS_USES_PLUGINS
+#include "utils/snd_utils.h"
+#endif
```

Comment:
So, do you need to recompile the library to use plugins? Isn't possible to determine at runtime if one should use plugins or real hw?

Reply:
Actually, our initial implementation had the same design where virtual(plugin) device was decided if the hw node is present or not and  if there is virtual node in configuration file. But as  discussed in PR for tinyalsa https://github.com/tinyalsa/tinyalsa/pull/137 , we got a feedback to handle it using compile time flag too so that it can be disabled for those who don't want plugin support.

Reply:
@rohkkumar ok.